diff -ruw a/src/libANGLE/Display.cpp b/src/libANGLE/Display.cpp
--- a/src/libANGLE/Display.cpp	2021-05-09 21:00:24.434527200 +0300
+++ b/src/libANGLE/Display.cpp	2021-05-09 21:17:23.252423900 +0300
@@ -164,6 +164,11 @@
 #if defined(ANGLE_ENABLE_OPENGL)
     if (angleDefaultEnv == "gl")
     {
+        return EGL_PLATFORM_ANGLE_TYPE_OPENGL_ANGLE;
+    }
+
+	if (angleDefaultEnv == "gles")
+    {
         return EGL_PLATFORM_ANGLE_TYPE_OPENGLES_ANGLE;
     }
 #endif
